helm-chart/openj9-jitserver-chart/README.md

```diff
@@ -0,0 +1,228 @@
+OMR      - 113e54219
+JCL      - 95bb504fbb based on jdk-11.0.8+10)
+```
+
+Find the corresponding image from [AdoptOpenJDK Docker Hub](https://hub.docker.com/_/adoptopenjdk) through keywords search of architecture, Java version and release. In this example, `0.21.0` release of Java 11 is used in the application image, thus the corresponding image tag is [`11.0.8_10-jdk-openj9-0.21.0`](https://hub.docker.com/layers/adoptopenjdk/library/adoptopenjdk/11.0.8_10-jdk-openj9-0.21.0/images/sha256-996bb5bca1ae3fbab7bcd6ed4b19776eb1e176727024ce8866a75536137949b7?context=explore). The image tag will be used in the following step.
+
+#### b. Install the JITServer using the selected image and the Helm Chart.
+Add OpenJ9 JITServer helm chart repository to the Helm client.
+``` bash
+helm repo add openj9 https://<updatewhenweknowtheurl>stable/openj9/repo/index.yaml
+helm repo update
+helm search repo openj9-jitserver-chart
+```
+
+Install OpenJ9 JITServer helm chart by following below command.
+``` bash
+helm install jitserver-release openj9-jitserver-chart
+```
+
+This chart deploys a JITServer instance with latest OpenJ9 Java 8 release on x86 platform by default. If your application uses a different version of Java, please include the option `--set image.tag="REPLACE_IMAGE_TAG"` in above command which overrides below section in `Values.yaml`. 
+```yaml
+image:
+  tag: 8-openj9
+```
+
+Java versions 8 and 11 are currently supported on x86, POWER and Z platforms.
+
+### 2. Enable your application to use JITServer.
+Once the JITServer is deployed, you can enable your application to use JITServer by adding the following JVM option as an environment variable into your Java application container.
+
+Service names can be found by `kubectl get service`, `<serverport>` is set to `38400` by default. 
+``` bash
+JAVA_OPTIONS = "-XX:+UseJITServer -XX:JITServerAddress=<servicename> -XX:JITServerPort=<serverport>"
+```
+
+`<servicename>` is set automatically by helm, and you may change `<serverport>` by including option `--set service.port="REPLACE_PORT_NUMBER"`. This option overrides below section in `Values.yaml`. 
+```yaml
+service:
+  port: 38400
+```
+
+Setting the `JAVA_OPTIONS` environment variable on a new or an already in production application depends on your application build/deployment pipeline. For more detailed instructions on how to redeploy your application with the above environment variable changes, please read through [this tutorial](./enable-jitserver.md). The tutorial shows how to enable JITServer technology for an example OpenLiberty Java application.
+
+## JITServer considerations and platform support
+### How to select a platform and a Java version 
+Platforms are handled by multi-arch images from AdoptOpenJDK docker hub, thus platform specific JITServer is deployed based on which platform the install is being done on. The JITServer Java version has to match the application Java version. For example, Java 8 applications need Java 8 JITServer.
```

Comment:
```suggestion
Platforms are handled by multi-arch images from AdoptOpenJDK docker hub, thus platform specific JITServer is deployed based on which platform the install is being done on. The JITServer Java version has to exactly match the application Java version. For example, Java 8 applications need Java 8 JITServer running on the exact same JDK version. (See details above)
```

Reply:
"running on the exact same JDK version" this is not accurate. The internal version of the JITServer technology must be matched between client and server. This can be found with an appropriate Xjit command line option:
```
docker run --rm -it -e OPENJ9_JAVA_OPTIONS="-Xjit:verbose={JITServer}" jitaas-server
JITServer is currently a technology preview. Its use is not yet supported.
#JITServer: JITServer version: 1.17.0
#JITServer: JITServer Server Mode. Port: 38400. Connection Timeout 30000ms
#JITServer: Started JITServer listener thread: 00000000001DFD00

JITServer is ready to accept incoming requests
```
Above, the internal JITServer version is 1.17.0. Every time we code-in a change that breaks the compatibility between the server and the client the minor number of the version is bumped. This can happen several times per release cycle, but definitely not with every change.
Because it's difficult to explain all this in a concise manner, to avoid confusion I would like to recommend people to use the same OpenJ9 release for both server and client.

Reply:
The reason I call out exact same match is also related to security fixes.  We either need to bump the jitserver version every time there's a security fix made to openj9 that might touch on the compiler's behaviour, or we need to mandate exact same version.

It would be bad if the jitserver version check passed when the application was deployed on a OpenJ9 version 0.xx.0 which included a security fix but the jitserver it was connected to hadn't been updated yet, leaving the application vulnerable.

Reply:
I agree with that argument and therefore I am ok with the proposed wording.